Test/dafny0/ArrayElementInitCompile.dfy

```diff
@@ -1,5 +1,8 @@
 // RUN: %dafny /compile:0 /print:"%t.print" /rprint:"%t.dprint" "%s" > "%t"
 // RUN: %dafny /noVerify /compile:4 /compileTarget:cs "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:java "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:js "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:go "%s" >> "%t"
```

Comment:
Would `%testDafnyForEachCompiler` work for this case now then?

Reply:
No. I spent 30+ minutes trying. using `%testDafnyForEachCompiler` causes a failure from the C++ compiler, which doesn't support "Subtraction of non-native type". The new test harness does not allow this, because this particular error is generated by an explicit `throw new ...` rather than a call through the expect method `UnsupportedFeatureError`. I tried to fix this and still keeping the description string "Subtraction of non-native type", but that causes the test harness not to recognize the failure in a different way. So, I tried giving up on the additional descriptive string, but then that shows another failure in the C++ compiler where it apparently outputs an identifier named `_#Func1`.

My conclusion: The new test harness does not handle C++ failures nicely and the C++ compiler does not report everything it doesn't support. I don't have an interest in fixing either of these problems in this PR.

Reply:
I appreciate the effort in trying to make it work! The issue is that `%testDafnyForEachCompiler` allows compilers to cleanly not support a feature, but they have to declare it in a specific way so we don't just allow arbitrary exceptions or failures. It sounds like you hit such a case in the C++ compiler - I went through and refactored all the obvious cases to use the right format, but I'm sure there are messy cases remaining, especially cases where it emits bad code instead of throwing an exception (often from code that was copied from another compiler without adjusting it :)

This is fine for now, we'll definitely be doing a pass over the test suite to convert them all to `%testDafnyForEachCompiler` at which point we'll fix the misbehaving compilers.